src/views/Profile/PersonalInfo/RegionCityHandler.tsx

```diff
@@ -0,0 +1,175 @@
+    try {
+      const res = await getRegionByCityId(id);
+      if (res.status === 200) {
+        const regionResponse = res.data;
+        setSelectRegions([regionResponse]);
+        setRegionForValue(regionResponse);
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    }
+  };
+
+  const setCitiesByRegionId = async (id: number) => {
+    try {
+      const res = await getCitiesByRegionId(id);
+      if (res.status === 200) {
+        const citiesResponse = res.data;
+        setSelectCities(citiesResponse);
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    }
+  };
+
+  const inputCityChangeHandler = (
+    _,
+    value: ICity | null,
+    inputIdentifier: string,
+  ) => {
+    if (!value) {
+      setErrorFields({
+        ...errorFields,
+        [inputIdentifier]: i18n.t(langTokens.admin.fieldCantBeEmpty),
+      });
+      setIsCitySelect(false);
+      setSelectRegions(null);
+      if (!regionForValue) {
+        setSelectCities(null);
+      }
+    } else {
+      setErrorFields({ ...errorFields, [inputIdentifier]: '' });
+      setNewAuthorValues({ ...newAuthorValues, [inputIdentifier]: value.id });
+      setRegionByCityId(value.id);
+      setIsCitySelect(true);
+    }
+  };
+
+  const inputRegionChangeHandler = (
+    _,
+    value: string | '',
+    inputIdentifier: string,
+  ) => {
+    if (!value) {
+      setErrorFields({
+        ...errorFields,
+        [inputIdentifier]: i18n.t(langTokens.admin.fieldCantBeEmpty),
+      });
+      setRegionForValue(null);
+      setSelectCities(null);
+      if (!isCitySelect) {
+        setSelectRegions(null);
+      }
+    } else {
+      const regionFromValue = regions.find((region) => region.name === value);
+      if (regionFromValue) {
+        setErrorFields({ ...errorFields, [inputIdentifier]: '' });
+        setNewAuthorValues({
+          ...newAuthorValues,
+          [inputIdentifier]: regionFromValue.id,
+        });
+        setCitiesByRegionId(regionFromValue.id);
+        setRegionForValue(regionFromValue);
+      }
+    }
+  };
+
+  return (
+    <Grid container spacing={2}>
+      <Grid item xs={6}>
+        <ErrorField
+          errorField={errorFields.regionId}
+          visitField={visitFields.region}
+        />
+        <Autocomplete
+          disablePortal
+          options={selectRegions || regions}
+          fullWidth
+          value={regionForValue}
+          getOptionLabel={(region: IRegion) => region.name}
+          getOptionSelected={(option, value) => option.id === value.id}
+          onInputChange={(_, value) =>
+            inputRegionChangeHandler(_, value, 'regionId')
+          }
+          renderInput={(params) => (
+            <TextField
+              variant="outlined"
+              className={classes.TextField}
+              required
+              // eslint-disable-next-line
+              {...params}
```

Comment:
typically we want `{...params}` at the very end or at the very beginning; putting it in the middle is not a technically wrong, but is not that typical and may cause a confusion if some props can be passed/overriden from outside and some cannot

Reply:
moved to the very end